src/common/snippets/include/snippets/runtime_configurator.hpp

```diff
@@ -122,12 +139,73 @@ class RuntimeConfigurator {
     void update_buffer_scratchpad_size(const lowered::LinearIRCPtr& linear_ir) const;
     /**
      * @brief Calculate data offsets of LinearIR and update these values in RuntimeConfig
+     * @param shapes shapes used in offsets computation
+     * @param layouts layouts used in offsets computation
+     */
+    void update_data_offsets(const std::vector<ov::snippets::VectorDims>& shapes,
+                             const std::vector<std::vector<size_t>>& layouts) const;
+    /**
+     * @brief Extract shapes from m_io_descs
      */
-    void update_data_offsets() const;
+    std::vector<ov::snippets::VectorDims> extract_shapes() const;
     /**
-     * @brief Update latest input shapes
+     * @brief Extract layouts from m_io_descs
      */
-    void update_latest_shapes();
+    std::vector<std::vector<size_t>> extract_layouts() const;
+
+    class ParallelWAOptimizer {
```

Comment:
This class is designed to work only for MHA patters, do you think it would be possible to generalize it to support eltwises?
In the meantime, should we rename it to smth like ParallelMHAOptimizer? Cause there is a lot of MHA-specific logic in there

Reply:
Yes, I think the optimizer can support eltwises. Moreover, it can be extended to support not only dimension splitting, but collapsing too. For now, I agree, let's rename it